test/testHelpers.hpp

```diff
@@ -389,9 +394,30 @@ ::testing::AssertionResult assertRefEq(std::string hA_name,
                                        const af::array &a, int expected);
 
 /// Checks if the C-API arrayfire function returns successfully
+/// If SKIP_UNSUPPORTED_TESTS is true then test will be skipped
+/// rather than failing
 ///
 /// \param[in] CALL This is the arrayfire C function
+#ifdef SKIP_UNSUPPORTED_TESTS
+#define ASSERT_SUCCESS(CALL)                          \
+    CHECK_SUPPORTED_ERR_RETURN = CALL;                \
+    CHECK_SUPPORTED(CHECK_SUPPORTED_ERR_RETURN);      \
+    ASSERT_EQ(AF_SUCCESS, CHECK_SUPPORTED_ERR_RETURN)
+#else
 #define ASSERT_SUCCESS(CALL) ASSERT_EQ(AF_SUCCESS, CALL)
+#endif
+
+/// Catches function unsupported exception thrown by C++ API
+/// arrayfire function and skips test if SKIP_UNSUPPORTED_TESTS is true
+///
+/// \param[in]  CALL This is the arrayfire C++ function
+#ifdef SKIP_UNSUPPORTED_TESTS
+#define FUNCTION_UNSUPPORTED \
```

Comment:
Not a fan of this approach to unsupported exceptions: Can you try using http://google.github.io/googletest/advanced.html#asserting-on-subroutines-with-an-exception

This should allow you to add a listener to catch these exceptions.

Reply:
The documentation you linked to seems to be for turning a failed assert into an exception, not for listening for an exception but I will investigate more to see if we can do what you suggest.